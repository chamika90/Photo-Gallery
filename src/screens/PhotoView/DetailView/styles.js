import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  detailContainer: {
    flexDirection: 'row',
    paddingVertical: 5,
  },
  detailTitle: {
    flex: 2,
    fontWeight: 'bold',
  },
  detailValue: {flex: 3},
});

export default styles;
