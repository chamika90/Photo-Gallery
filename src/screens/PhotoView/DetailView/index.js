import React from 'react';
import {View, Text, TouchableOpacity, Linking} from 'react-native';

import BottomModal from 'components/BottomModal';
import styles from './styles';

const DetailRow = ({title, value}) => {
  return (
    <View style={styles.detailContainer}>
      <Text style={styles.detailTitle}>{title}</Text>
      {(title === 'Web' && (
        <Text
          style={styles.linkText}
          onPress={() => {
            Linking.openURL(value);
          }}>
          {value}
        </Text>
      )) || <Text style={styles.detailValue}>{value}</Text>}
    </View>
  );
};

const DetailView = ({isVisible, item, onClose, onDownloadPress}) => {
  return (
    <BottomModal isVisible={isVisible} onClose={() => onClose()}>
      <View>
        <DetailRow title={'Author'} value={item.author} />
        <DetailRow title={'Web'} value={item.url} />
        <TouchableOpacity
          style={styles.downloadButton}
          onPress={() => onDownloadPress()}>
          <Text style={styles.downloadText}>Download</Text>
        </TouchableOpacity>
      </View>
    </BottomModal>
  );
};

export default DetailView;
