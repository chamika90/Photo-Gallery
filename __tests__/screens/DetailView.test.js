import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';

import DetailView from 'screens/PhotoView/DetailView';

describe('DetailView', () => {
  it('should match snapshot', async () => {
    const {toJSON} = render(
      <DetailView
        isVisible={true}
        item={{
          author: 'Alejandro Escamilla',
          download_url: 'https://picsum.photos/id/1/5616/3744',
          height: 3744,
          id: '1',
          url: 'https://unsplash.com/photos/LNRyGwIJr5c',
          width: 5616,
        }}
        onClose={() => {
          console.log('on close modal');
        }}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
