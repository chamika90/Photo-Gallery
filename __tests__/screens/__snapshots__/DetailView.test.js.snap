// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailView should match snapshot 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
        "margin": 0,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "rgba(0, 0, 0, 0.1)",
          "borderTopLeftRadius": 15,
          "borderTopRightRadius": 15,
          "justifyContent": "center",
          "padding": 22,
        }
      }
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "flex": 2,
                  "fontWeight": "bold",
                }
              }
            >
              Author
            </Text>
            <Text
              style={
                Object {
                  "flex": 3,
                }
              }
            >
              Alejandro Escamilla
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "flex": 2,
                  "fontWeight": "bold",
                }
              }
            >
              Url
            </Text>
            <Text
              style={
                Object {
                  "flex": 3,
                }
              }
            >
              https://unsplash.com/photos/LNRyGwIJr5c
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "flex": 2,
                  "fontWeight": "bold",
                }
              }
            >
              Download Url
            </Text>
            <Text
              style={
                Object {
                  "flex": 3,
                }
              }
            >
              https://picsum.photos/id/1/5616/3744
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
